swap/src/bin/swap.rs

```diff
@@ -222,8 +170,135 @@ async fn main() -> Result<()> {
             // Print the table to stdout
             table.printstd();
         }
-        Options::Resume { .. } => todo!("implement this"),
+        Command::Resume {
+            swap_id,
+            bitcoind_url,
+            bitcoin_wallet_name,
+            monero_wallet_rpc_url,
+            listen_addr,
+        } => {
+            let db_swap = db.get_state(swap_id)?;
+
+            if let Ok(alice_state) = AliceState::try_from(db_swap.clone()) {
```

Comment:
Instead of `if let _ else if let`, match on `db_swap` (alice or bob) and error out if the try_from errors out.

Reply:
Was not exactly sure how to achieve this, let's please do a small pairing session tomorrow and create a follow-up PR!

Reply:
```
match db_swap {
	Alice => { db_swap.try_into()?;...}
	Bob => { db_swap.try_into()?;...}
}
```